JWBestPracticeApps/JWFriendlyObstructions/JWFriendlyObstructionViewController.m

```diff
@@ -31,7 +32,8 @@ - (void)viewDidLoad {
     self.playIcon = [UIImage imageNamed:playIconName];
     self.pauseIcon = [UIImage imageNamed:pauseIconName];
     [self addCustomControl];
-    [self.player.experimentalAPI registerFriendlyAdObstruction:self.playbackToggle];
+    JWFriendlyAdObstructions *friendlyAdObstructions = [[JWFriendlyAdObstructions alloc] initWithPlayer:self.player];
```

Comment:
Since no-one is retaining the friendlyAdObstructions object it will be deallocated by the ARC. I suggests to add it as a strong property in the view controller.

Reply:
Hi David, I changed it. Please let me know what you think when you get the chance.